spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/MongoDbContainerTest.java

```diff
@@ -40,53 +42,68 @@
  *
  * @author Oleg Zhurakousky
  * @author Xavier Padro
  * @author Artem Bilan
  * @author David Turanski
+ * @author Artem Vozhdayenko
  *
- * @since 2.1
+ * @since 6.0
  */
-public abstract class MongoDbAvailableTests {
-
-	@Rule
-	public MongoDbAvailableRule mongoDbAvailableRule = new MongoDbAvailableRule();
-
-	public static final MongoDatabaseFactory MONGO_DATABASE_FACTORY =
-			new SimpleMongoClientDatabaseFactory(
-					MongoClients.create(
-							MongoClientSettings.builder().uuidRepresentation(UuidRepresentation.STANDARD).build()),
-					"test");
-
-	public static final ReactiveMongoDatabaseFactory REACTIVE_MONGO_DATABASE_FACTORY =
-			new SimpleReactiveMongoDatabaseFactory(
-					com.mongodb.reactivestreams.client.MongoClients.create(
-							MongoClientSettings.builder().uuidRepresentation(UuidRepresentation.STANDARD).build()),
-					"test");
-
-	protected MongoDatabaseFactory prepareMongoFactory(String... additionalCollectionsToDrop) {
-		cleanupCollections(MONGO_DATABASE_FACTORY, additionalCollectionsToDrop);
-		return MONGO_DATABASE_FACTORY;
+@Testcontainers(disabledWithoutDocker = true)
+public interface MongoDbContainerTest {
+	GenericContainer<?> MONGO_CONTAINER = new GenericContainer<>("mongo:5.0.9")
+			.withExposedPorts(27017);
+
+	@BeforeAll
+	static void startContainer() {
+		MONGO_CONTAINER.start();
+	}
+
+	static MongoDatabaseFactory createMongoDbFactory() {
+		return new SimpleMongoClientDatabaseFactory(
+				MongoClients.create(
+						MongoClientSettings.builder()
+								.applyConnectionString(new ConnectionString(
+										"mongodb://localhost:" + MONGO_CONTAINER.getFirstMappedPort()))
+								.uuidRepresentation(UuidRepresentation.STANDARD).build()),
+				"test");
+	}
+
+	static ReactiveMongoDatabaseFactory createReactiveMongoDbFactory() {
+		return new SimpleReactiveMongoDatabaseFactory(
+				com.mongodb.reactivestreams.client.MongoClients.create(
+						MongoClientSettings.builder().applyConnectionString(new ConnectionString(
+										"mongodb://localhost:" + MONGO_CONTAINER.getFirstMappedPort()))
+								.uuidRepresentation(UuidRepresentation.STANDARD).build()),
+				"test");
+	}
+
+	static void prepareMongoData(MongoDatabaseFactory mongoDatabaseFactory, String... additionalCollectionsToDrop) {
+		cleanupCollections(mongoDatabaseFactory, additionalCollectionsToDrop);
 	}
 
-	protected ReactiveMongoDatabaseFactory prepareReactiveMongoFactory(String... additionalCollectionsToDrop) {
-		cleanupCollections(REACTIVE_MONGO_DATABASE_FACTORY, additionalCollectionsToDrop);
-		return REACTIVE_MONGO_DATABASE_FACTORY;
+	static void prepareReactiveMongoData(ReactiveMongoDatabaseFactory mongoDatabaseFactory, String... additionalCollectionsToDrop) {
+		cleanupCollections(mongoDatabaseFactory, additionalCollectionsToDrop);
 	}
 
-	protected void cleanupCollections(ReactiveMongoDatabaseFactory mongoDbFactory,
+	static void cleanupCollections(ReactiveMongoDatabaseFactory MONGO_DATABASE_FACTORY,
```

Comment:
I don't think this change was intentional.
Please, bring the parameter name back.

Reply:
Shame..